L1Trigger/L1TCalorimeter/interface/AccumulatingSort.h

```diff
@@ -0,0 +1,50 @@
+#include <vector>
+#include <list>
+#include <stdint.h>
+
+
+template < typename T >
+class AccumulatingSort
+{
+
+private:
+    std::vector < std::list < T > > mSortArrays;
+    uint32_t mSize;
+
+private:
+    void AccumulatorUnit( const std::list< T >& aInput , T& aAcc , std::list< T >& aTail )
+    {
+        aTail.clear();
+
+        bool lAccInserted( false );
+
+        for( typename std::list< T >::const_iterator lIt( aInput.begin() ) ; lIt != aInput.end() ; ++lIt )
```

Comment:
Pls reserve the memory needed for aTail from the size of aInput
(the erase can be avoided too, but that is less important)